### PR TITLE
Add signed certificates and keys in echo-server sample

### PR DESCRIPTION
**Pull request by rveerama1**
Current TLS test configuration has only certs and keys. Added signed certificates, keys and CA to test echo-client from net-tools.
NET_SOCKETS_SOCKOPT_TLS should be enabled to use these certs.

Same CA, certs and keys will used to test TLS in Testsuite and in net-tools.

Reference net-tools PR : https://github.com/zephyrproject-rtos/net-tools/pull/39

### CODE REVIEW

File: samples/net/sockets/echo_server/src/tcp.c [hunk 1]
```diff
@@ -220,7 +220,7 @@ static int process_tcp(struct data *data)
 			&client_addr_len);
 	if (client < 0) {
 		LOG_ERR("%s accept error (%d)", data->proto, -errno);
-		return -errno;
+		return 0;
```
Comment (rlubos) [inline samples/net/sockets/echo_server/src/tcp.c]: Is this change intended?

**Review by rlubos — COMMENTED** [review]
The already existing keys also originated from `net-tools` repo, so it was possible to test the `echo_server` with the `echo_client` from net tools. Those certs were self-signed tough, I guess there was a need for a different setup (with a proper CA), so LGTM, provided the one nit is clarified.

File: samples/net/sockets/echo_server/Kconfig [hunk 2]
```diff
@@ -58,4 +58,12 @@ config NET_SAMPLE_PSK_HEADER_FILE
 	  Name of a header file containing a
 	  pre-shared key.
 
+config NET_SAMPLE_CERTS_WITH_SC
+	bool "Signed certificates"
+	default n
+	depends on NET_SOCKETS_SOCKOPT_TLS
+	help
+	 Enable this flag, if you are interested to run this
```
Comment (jukkar) [inline samples/net/sockets/echo_server/Kconfig]: Small style nit here, the help text should be intended by two chars, now there is one

File: samples/net/sockets/echo_server/Kconfig [hunk 3]
```diff
@@ -58,4 +58,12 @@ config NET_SAMPLE_PSK_HEADER_FILE
 	  Name of a header file containing a
 	  pre-shared key.
 
+config NET_SAMPLE_CERTS_WITH_SC
+	bool "Signed certificates"
+	default n
```
Comment (jukkar) [inline samples/net/sockets/echo_server/Kconfig]: `default n` can be removed as the setting is not enabled by default

**Review by jukkar — COMMENTED** [review]
Looks ok, small style issue

File: samples/net/sockets/echo_server/src/tcp.c (hunk 1 above)
Comment (rveerama1) [reply samples/net/sockets/echo_server/src/tcp.c]: Yes, Issue noticed with Testsuite. When TLS handshake fails, it results in accept() errors, which causes echo_server to stop. And consecutive tests fails to run. If there is an handshake / accept error echo_server should go back and start listening to accept new incoming connections.

File: samples/net/sockets/echo_server/Kconfig (hunk 2 above)
Comment (rveerama1) [reply samples/net/sockets/echo_server/Kconfig]: Done.

File: samples/net/sockets/echo_server/Kconfig (hunk 3 above)
Comment (rveerama1) [reply samples/net/sockets/echo_server/Kconfig]: Done.

File: samples/net/sockets/echo_server/src/tcp.c (hunk 1 above)
Comment (rlubos) [reply samples/net/sockets/echo_server/src/tcp.c]: Ok, thanks for clarifying, I guess it's fine then.

**Review by rlubos — APPROVED** [review]

File: samples/net/sockets/echo_server/src/tcp.c (hunk 1 above)
Comment (pfalcon) [reply samples/net/sockets/echo_server/src/tcp.c]: > If there is an handshake / accept error echo_server should go back and start listening to accept new incoming connections.

Just correction/clarification on terminology - it can go back and keep accept'ing new connections. Because indeed, accept() error may be transient (may be not) in the sense of the whole system, i.e. if one accept() fails, next may still succeed. A typical example of this would be no free file descriptor - the accept() has no choice but fail, but next time, some other part of system could free up fd, and accept() would succeed. Error in accept() doesn't change listening status of a socket, i.e. there's not need to call listen() again.

Whether this change *should* be done is a different question. It depends on what's meant by:

> When TLS handshake fails

Why does it fail? Generally, samples are written with "fail fast" approach, to make any failure visible, instead of employing sophisticated (or not) error handling.

Comment (rveerama1) [reply samples/net/sockets/echo_server/src/tcp.c]: > samples are written with "fail fast" approach, to make any failure visible,

That's true. But particularly this sample used to test network stack against Testsuite. That's the reason tuning the error handling. Hope it didn't harm.

Comment (rveerama1) [reply samples/net/sockets/echo_server/src/tcp.c]: > Why does it fail?

There are some tests, testsuite trying to fail the handshake, expecting TCP connection close. This is fine, but echo_server stopped and unable to continue further.

Comment (pfalcon) [reply samples/net/sockets/echo_server/src/tcp.c]: > But particularly this sample used to test network stack against Testsuite. That's the reason tuning the error handling.

Makes sense to me, thanks for explanation.

**Review by jukkar — APPROVED** [review]

**Comment by rveerama1** [comment]
To start CI
